databroker/databroker.py

```diff
@@ -505,3 +542,19 @@ def get_fields(header):
         for field in desc['data_keys'].keys():
             fields.add(field)
     return fields
+
+def _check_fields_exist(fields, headers):
+    all_fields = set()
+    for header in headers:
+        all_fields.update(header['start'])
```

Comment:
today I learned that you can update a set with a dictionary and it just grabs the keys out of the dict. neat!